conn.go

```diff
@@ -374,3 +414,29 @@ func (conn *Conn) subscribed(topic string) bool {
 	}
 	return false
 }
+
+func (conn *Conn) SetClosed() {
+	if wasClosed := atomic.CompareAndSwapInt32(&conn.closed, 0, 1); wasClosed {
+		conn.notifyOnCloseError()
+	}
+}
+
+func (conn *Conn) Closed() bool {
+	return atomic.LoadInt32(&conn.closed) == 1
+}
+
+func (conn *Conn) checkIO(err error) {
+	if err == nil {
+		return
+	}
+
+	if e, ok := err.(net.Error); ok && !e.Timeout() {
```

Comment:
this won't work if the error has been wrapped with `xerrors.Errorf("foo: %w", e)`.

the following should work:
```go
var e net.Error
if xerrors.Is(err, e); e != nil && !e.Timeout() {
    conn.SetClosed()
}
```

Reply:
you probably mean `xerrors.As(err, &e)`. But I got it, thanks.

Reply:
right. :)
(note self: never review code before 3rd coffee)